### PR TITLE
fix(#147): CUI-7438 [Accessibility] Calendar keyboard navigation by Arrow, (Alt + ) PageUp/Down, etc

### PR DESCRIPTION
**Pull request by majornista**
## Description

1. User should be able to navigate Calendar grid by week using ArrowUp/ArrowDown, by month using PageUp/PageDown, and by year using Alt+PageUp/Alt+PageDown.
2. Calendar should account for the differences in length between months and navigate to the last day of the next or previous month when the current date does not appear in the next or previous month.
3. DateTime should account for Daylight Savings Time when calculating the difference between dates when one of the dates falls in Daylight Savings Time but the other date does not.
4. We should stopPropagation when navigating using the keyboard to prevent calendar navigation in something like a DatePicker from bubbling up to be handled by the application.

## Related Issue
Relates to: https://jira.corp.adobe.com/browse/CQ-4310946 and https://jira.corp.adobe.com/browse/CUI-7438

## Motivation and Context
Fix for keyboard Accessibility in Calendar and Datepicker controls.

## How Has This Been Tested?
Tested against Coral-Spectrum example pages for Calendar and Datepicker components

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

**Comment by majornista** [comment]
Could I get a review? @icaraps

**Comment by icaraps** [comment]
cc @Pareesh @indra2gurjar

**Comment by majornista** [comment]
@Pareesh, @indra2gurjar Can anyone please review this?

**Comment by majornista** [comment]
@Pareesh, @indra2gurjar Can anyone please review this?

**Comment by Pareesh** [comment]
@ShreeSub Please review this PR, related to accessibility.

File: coral-component-calendar/src/scripts/Calendar.js [hunk 1]
```diff
@@ -779,20 +797,36 @@ class Calendar extends BaseFormField(BaseComponent(HTMLElement)) {
...
+          newMoment = this.min;
+        }
+        else if (this._isAfterMax(newMoment)) {
+          newMoment = this.max;
+        }
+        newMoment = new DateTime.Moment(newMoment);
         difference = Math.abs(new DateTime.Moment(currentActive).diff(newMoment, 'days'));
         this._getToNewMoment(direction, operator, difference);
         this._setActiveDescendant();
+        return;
```
Comment (Pareesh) [inline coral-component-calendar/src/scripts/Calendar.js:817]: @majornista , move the common code out of if-else statement

File: coral-component-calendar/src/scripts/Calendar.js [hunk 2]
```diff
@@ -28,9 +28,11 @@ function isDateInRange(date, startDate, endDate) {
 
   if (startDate === null && endDate === null) {
     return true;
-  } else if (startDate === null) {
+  }
```
Comment (Pareesh) [inline coral-component-calendar/src/scripts/Calendar.js:31]: "1tbs" braces-style appears more readability(my-thought).

Comment (majornista) [reply coral-component-calendar/src/scripts/Calendar.js:31]: ESLint throws warnings in this project:

```
/github.com/adobe/coral-spectrum/coral-component-calendar/src/scripts/Calendar.js
  353:5   warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  373:5   warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  585:23  warning  Method '_setActiveDescendant' has a complexity of 17. Maximum allowed is 15  complexity
  601:13  warning  Blocks are nested too deeply (5). Maximum allowed is 4                       max-depth
  603:13  warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  607:9   warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  639:7   warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  729:13  warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  732:11  warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  815:9   warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  862:5   warning  Closing curly brace appears on the same line as the subsequent block         brace-style
  879:5   warning  Closing curly brace appears on the same line as the subsequent block         brace-style 
 ```

Comment (Pareesh) [reply coral-component-calendar/src/scripts/Calendar.js:31]: @majornista 
This maybe due to https://github.com/adobe/coral-spectrum/blob/master/coral-gulp/configs/eslint.conf.js#L8 . This review comment was my personal opinion. I found "1tbs" more readable.  Feel free to resolve it, as per your convenience.

File: coral-component-calendar/src/scripts/Calendar.js [hunk 3]
```diff
@@ -964,31 +1008,43 @@ class Calendar extends BaseFormField(BaseComponent(HTMLElement)) {
...
   _onCtrlPageUpKey(event) {
     event.preventDefault();
+    event.stopPropagation();
     this._gotoPreviousOrNextTimeUnit(timeUnit.YEAR, false);
     this._validateCalendar();
   }
 
+  /** @ignore */
+  _onBodyFocus() {
+    if (!!this._value) {
```
Comment (Pareesh) [inline coral-component-calendar/src/scripts/Calendar.js:1042]: use transform.boolean to tranform value to boolean.

File: coral-component-calendar/src/scripts/Calendar.js (hunk 2 above)
Comment (majornista) [reply coral-component-calendar/src/scripts/Calendar.js:31]: I think following the preferred convention of coral-spectrum would be better.

File: coral-datetime/src/scripts/DateTime.js [hunk 4]
```diff
@@ -280,20 +296,49 @@ class DateTime {
...
     }
 
     return this;
   }
 
+
+  /**
+   @see https://momentjs.com/docs/#/displaying/days-in-month/
+  */
+  daysInMonth() {
```
Comment (Pareesh) [inline coral-datetime/src/scripts/DateTime.js:328]: Couldn't find its usage, if unused code please remove it.

**Review by Pareesh — APPROVED** [review]

File: coral-datetime/src/scripts/DateTime.js (hunk 4 above)
Comment (majornista) [reply coral-datetime/src/scripts/DateTime.js:328]: https://github.com/adobe/coral-spectrum/pull/148/files#diff-1814d617222a9f2c5f83f21f9b4e4877e0dbbbe630a731975cab53e5bbedfabaR1000

**Pull request merged**